### PR TITLE
Ncnews 5

### PR DESCRIPTION
**Pull request by bathcoder**
1. Created /api/articles/patch to increase vote count
2. Created error testing for: 
invalid endpoint data type
valid endpoint data type but wrong information
invalid input key 
invalid input property
empty input object

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -72,3 +72,71 @@ describe("/api/article/:article_id", () => {
...
+  test("status 200: responds with the updated article", () => {
+    return request(app)
+      .patch("/api/articles/3")
+      .send(voteUpdate)
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article).toEqual({
+          article_id: 3,
+          votes: 500,
+          ...body.article,
```
Comment (dappernerddesigns) [inline __tests__/app.test.js]: If you are just checking ID and votes, consider objectContaining matcher, then it is more flexible for more keys. Without an odd spread going on.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -72,3 +72,71 @@ describe("/api/article/:article_id", () => {
...
+  test.only("status 404: Returns a Route Not found message when given endpoint of correct type but is otherwise invalid", () => {
+    return request(app)
+      .patch("/api/articles/9999")
+      .send(voteUpdate)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Route Not Found");
+      });
+  });
+  test("status 422: returns an incorrect input message when sent invalid data key", () => {
```
Comment (dappernerddesigns) [inline __tests__/app.test.js:112]: This could be a 400, but after some reading myself, I like it. Nice find :)

**Review by dappernerddesigns — APPROVED** [review]
I would consider a matcher for objectContaining, but nice selection of tests and all errors handled.

File: __tests__/app.test.js (hunk 1 above)
Comment (bathcoder) [reply __tests__/app.test.js]: ok good idea

File: __tests__/app.test.js (hunk 2 above)
Comment (bathcoder) [reply __tests__/app.test.js:112]: yes, its contentious, theres is disagreement on this when i looked into it. But it seems applicable and i wanted to try a different error :).

**Pull request merged**